### PR TITLE
Bug 1179353 - Implement a stable screen automatic brightness algorithm.

### PR DESCRIPTION
**Pull request by cervantesyu**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=676387584922ad9e7ce4bac244adcd946e6915bc

File: apps/system/js/screen_auto_brightness.js [hunk 1]
```diff
@@ -15,11 +15,262 @@
...
+    this._currentBrightness = navigator.mozPower.screenBrightness;
+    this._state = this.STATE_COOLING_DOWN;
+  };
+
+  // reinitialize to start state
+  ScreenAutoBrightness.prototype.reset = function() {
+    this._previousBrightnessValues = new ScreenAutoBrightnessValues();
+
+    if (this._delayTimerID) {
+        clearTimeout(this._delayTimerID);
```
Comment (timdream) [inline apps/system/js/screen_auto_brightness.js]: extra 2 spaces.

File: apps/system/js/screen_auto_brightness.js [hunk 2]
```diff
@@ -15,11 +15,262 @@
...
+      this.doAutoAdjustOnMovingUp();
+      break;
+    case this.STATE_MOVING_DOWN:
+      this.doAutoAdjustOnMovingDown();
+      break;
+    case this.STATE_PAUSED:
+      this.doAutoAdjustOnPaused();
+      break;
+    default:
+      console.log('ScreenAutoBrightness: ' +
```
Comment (timdream) [inline apps/system/js/screen_auto_brightness.js]: `throw new Error(...);` and drop the `break` below.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7ac68c5eb93be533d0d9a5c88eb93aff0f5027c7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f1e7596804372e5083fbcd91debc48b9080664c5

**Pull request merged**